tests/functional/test_products.py

```diff
@@ -0,0 +1,69 @@
+import requests
+url = "http://localhost:5000/"
+
+
+def test_products_get(first_product):
+    # Test a valid product retrieve
+    r = requests.get(url + "API/product/1")
+    assert r.status_code == 200
+    assert r.json() == first_product
+
+    # Test an invalid product retrieve
+    r = requests.get(url + "API/product/0")
+    assert r.status_code == 500  # 404
+
+
+def test_products_list_get(products_json):
+    # Test retrieving all products
+    r = requests.get(url + "API/products")
+    assert r.status_code == 200
+    assert r.json() == products_json
+
+
+def test_add_product_post(user_auth):
+    json = {
+        'name': 'Nintendo Switch',
+        'category': 'Consolas y Videojuegos',
+        'price': 150,
+        'condition': 'Casi nuevo',
+        'description': 'La vendo porque apenas la uso y necesito algo de dinero',
+        'shipment': False
+    }
+
+    # Test empty fields
+    r = requests.post(url + "API/catalog/add/pepe432@gmail.com", auth=user_auth)
```

Comment:
Esto no funciona ya que en el post hay que poner explícitamente el body. 
```suggestion
    r = requests.post(url + "API/catalog/add/pepe432@gmail.com", json=json, auth=user_auth)

Reply:
Creo que esto también está relacionado con la versión del `requests` ya que no me daba ningún problema y hacía el assert correctamente. Ya está cambiado.